### PR TITLE
Avoid repeating the entity's name on each endpoint

### LINKED ISSUE #23553: Backend: Avoid repeating the entity's name on each endpoint
##### **Overview of the feature request**
The Resource files shouldn't repeat the entity's name on each endpoint as it's redundant.

##### **Motivation for or Use Case**
The entity's name should be at the start of the class so you don't need to repeat it every time you add a new endpoint.

##### **Related issues or PR**
Searched, none found.

- [X] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by amatosg**
Closes #23553 Remove entity's name from each endpoint
---
With this changes, each endpoint doesn't has to have the entity's name on each endpoint.

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/entity/src/main/java/package/web/rest/_EntityClass_Resource.java.ejs [hunk 1]
```diff
@@ -163,7 +163,7 @@ public class <%= entityClass %>Resource {
      * @return the {@link ResponseEntity} with status {@code 201 (Created)} and with body the new <%= instanceName %>, or with status {@code 400 (Bad Request)} if the <%= entityInstance %> has already an ID.
      * @throws URISyntaxException if the Location URI syntax is incorrect.
      */
-    @PostMapping("/<%= entityApiUrl %>")
+    @PostMapping("")
```
Comment (mshima) [inline generators/server/templates/entity/src/main/java/package/web/rest/_EntityClass_Resource.java.ejs:166]: Maybe
```suggestion
    @PostMapping("/")
```

File: generators/server/templates/entity/src/main/java/package/web/rest/_EntityClass_Resource.java.ejs [hunk 2]
```diff
@@ -360,7 +360,7 @@ public class <%= entityClass %>Resource {
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and the list of <%= entityInstancePlural %> in body.
      */
 <%_ if (reactive) { _%>
-    @GetMapping(value = "/<%= entityApiUrl %>", produces = MediaType.APPLICATION_JSON_VALUE)
+    @GetMapping(value = "", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (mshima) [inline generators/server/templates/entity/src/main/java/package/web/rest/_EntityClass_Resource.java.ejs:363]:
```suggestion
    @GetMapping(value = "/", produces = MediaType.APPLICATION_JSON_VALUE)
```

File: generators/server/templates/entity/src/main/java/package/web/rest/_EntityClass_Resource.java.ejs [hunk 3]
```diff
@@ -374,7 +374,7 @@ public class <%= entityClass %>Resource {
      * {@code GET  /<%= entityApiUrl %>} : get all the <%= entityInstancePlural %> as a stream.
      * @return the {@link Flux} of <%= entityInstancePlural %>.
      */
-    @GetMapping(value = "/<%= entityApiUrl %>", produces = MediaType.APPLICATION_NDJSON_VALUE)
+    @GetMapping(value = "", produces = MediaType.APPLICATION_NDJSON_VALUE)
```
Comment (mshima) [inline generators/server/templates/entity/src/main/java/package/web/rest/_EntityClass_Resource.java.ejs:377]:
```suggestion
    @GetMapping(value = "/", produces = MediaType.APPLICATION_NDJSON_VALUE)
```

**Comment by mshima** [comment]
I've remembered I tried this before but it won't work.
The search api is different `api/_search/entity-name`.
Is this api compliant with conventions?

**Comment by amatosg** [comment]
> The search api is different api/_search/entity-name.

Why change the the behavior? Is that the only way to do it? It would make more sense to have it `api/entity-name/_search`

**Comment by mshima** [comment]
> > The search api is different api/_search/entity-name.
> 
> Why change the the behavior? Is that the only way to do it? It would make more sense to have it `api/entity-name/_search`

I agree with the change before final v8.
I prefer to be handled in another PR if someone is willing to do.

**Comment by amatosg** [comment]
wouldn't use 2 separate PRs break the tests?

**Comment by mshima** [comment]
@amatosg fill free to change here, another PR would be more concise.

**Comment by amatosg** [comment]
@mshima any idea what else needs to be changed? tests are failing but I cannot find the solution

File: generators/server/templates/src/main/java/_package_/_entityPackage_/_partials_entity_/get_all_template.ejs [hunk 4]
```diff
@@ -62,7 +62,7 @@ _%>
      * @param criteria the criteria which the requested entities should match.
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and the count in body.
      */
-    @GetMapping("/<%= entityApiUrl %>/count")
+    @GetMapping("count")
```
Comment (mshima) [inline generators/server/templates/src/main/java/_package_/_entityPackage_/_partials_entity_/get_all_template.ejs:65]:
```suggestion
    @GetMapping("/count")
```

File: generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs [hunk 5]
```diff
@@ -130,7 +130,7 @@ import java.util.stream.StreamSupport;
  * REST controller for managing {@link <%= entityAbsoluteClass %>}.
  */
 @RestController
-@RequestMapping("/api")
+@RequestMapping("/api/<%= entityApiUrl %>/")
```
Comment (mshima) [inline generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs:133]:
```suggestion
@RequestMapping("/api/<%= entityApiUrl %>")
```

File: generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs [hunk 6]
```diff
@@ -217,7 +217,7 @@ public class <%= entityClass %>Resource {
      * or with status {@code 500 (Internal Server Error)} if the <%= instanceName %> couldn't be updated.
      * @throws URISyntaxException if the Location URI syntax is incorrect.
      */
-    @PutMapping("/<%= entityApiUrl %>/{<%= primaryKey.name %>}")
+    @PutMapping("{<%= primaryKey.name %>}")
```
Comment (mshima) [inline generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs:220]:
```suggestion
    @PutMapping("/{<%= primaryKey.name %>}")
```

File: generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs [hunk 7]
```diff
@@ -283,7 +283,7 @@ public class <%= entityClass %>Resource {
      * or with status {@code 500 (Internal Server Error)} if the <%= instanceName %> couldn't be updated.
      * @throws URISyntaxException if the Location URI syntax is incorrect.
      */
-    @PatchMapping(value = "/<%= entityApiUrl %>/{<%= primaryKey.name %>}", consumes = {"application/json", "application/merge-patch+json"})
+    @PatchMapping(value = "{<%= primaryKey.name %>}", consumes = {"application/json", "application/merge-patch+json"})
```
Comment (mshima) [inline generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs:286]:
```suggestion
    @PatchMapping(value = "/{<%= primaryKey.name %>}", consumes = {"application/json", "application/merge-patch+json"})
```

File: generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs [hunk 8]
```diff
@@ -394,7 +394,7 @@ public class <%= entityClass %>Resource {
      * @param id the id of the <%= instanceName %> to retrieve.
      * @return the {@link ResponseEntity} with status {@code 200 (OK)} and with body the <%= instanceName %>, or with status {@code 404 (Not Found)}.
      */
-    @GetMapping("/<%= entityApiUrl %>/{id}")
+    @GetMapping("{id}")
```
Comment (mshima) [inline generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs:397]:
```suggestion
    @GetMapping("/{id}")
```

File: generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs [hunk 9]
```diff
@@ -410,7 +410,7 @@ public class <%= entityClass %>Resource {
      * @param id the id of the <%= instanceName %> to delete.
      * @return the {@link ResponseEntity} with status {@code 204 (NO_CONTENT)}.
      */
-    @DeleteMapping("/<%= entityApiUrl %>/{id}")
+    @DeleteMapping("{id}")
```
Comment (mshima) [inline generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs:413]:
```suggestion
    @DeleteMapping("/{id}")
```

File: generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs [hunk 10]
```diff
@@ -444,6 +444,6 @@ public class <%= entityClass %>Resource {
   <%_ } _%>
      * @return the result of the search.
      */
-    @GetMapping("/_search/<%= entityApiUrl %>")<%- include('../../_partials_entity_/search_template', {viaService}); -%>
+    @GetMapping("_search")<%- include('../../_partials_entity_/search_template', {viaService}); -%>
```
Comment (mshima) [inline generators/server/templates/src/main/java/_package_/_entityPackage_/web/rest/_entityClass_Resource.java.ejs:447]:
```suggestion
    @GetMapping("/_search")<%- include('../../_partials_entity_/search_template', {viaService}); -%>
```

**Comment by mshima** [comment]
Need to adjust more apis.
At least https://github.com/jhipster/generator-jhipster/blob/e68a3a4fb192cd382c59f78ef8dfec62a23192e1/generators/server/templates/src/main/java/_package_/web/rest/PublicUserResource.java.ejs#L183

**Comment by amatosg** [comment]
@mshima looks like your changes were enough. Thank you!

**Review by DanielFran — APPROVED** [review]

**Comment by DanielFran** [comment]
@amatosg Can you take a look in the [hipster](https://github.com/jhipster/jhipster.github.io) if there is any needed update in the documentation?

**Pull request merged**